src/client/Settings.tsx

```diff
@@ -43,6 +45,12 @@ export const Settings = ({
     [],
   );
 
+  const usernameRef = useRef<HTMLInputElement>(null);
+  const handleUsernameChange = () => {
+    const usernameValue = usernameRef.current?.value;
+    username = usernameValue;
```

Comment:
```suggestion
    setUsername(usernameValue);
```

Reply:
The implementation of `setUsername` will likely feed into the reducer, something like:

```
const setUsername = useCallback((username) => {
  dispatch({type:"set_username", username});
```

Reply:
Let's focus on this after the main functionality is working.